modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/LiferayOAuthDataProvider.java

```diff
@@ -1001,6 +1005,15 @@ else if (_oAuth2ProviderConfiguration.
 
 				clientGrantTypes.add(OAuthConstants.CLIENT_CREDENTIALS_GRANT);
 			}
+			else if (_oAuth2ProviderConfiguration.allowJWTBearerGrant() &&
+					 (allowedGrantType == GrantType.JWT_BEARER)) {
+
+				clientGrantTypes.add(Constants.JWT_BEARER_GRANT);
+				clientGrantTypes.add(
+					HttpUtils.urlEncode(
+						Constants.JWT_BEARER_GRANT,
+						StandardCharsets.UTF_8.name()));
```

Comment:
@arthurchan35 why are we adding this grant type with and without URL encoding?

Reply:
Hey @stian-sigvartsen 
It's a common practice. 
For custom grant types like JWT bearer, it's often to have both encoded and un-encoded versions as the custom type has characters like ":" that OAuth server cannot know for certain if a request contains the encoded or un-encoded version because both are allowed in http.